### PR TITLE
change group 

### PR DESCRIPTION
**Pull request by Madhu-1**
if the gluster group is not present on the
a system where glusterd2 is supposed to run,
set Current user GID as UID for auth file

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@aravindavk please review

File: glusterd2/gdctx/global.go [hunk 1]
```diff
@@ -94,18 +94,33 @@ func GenerateLocalAuthToken() error {
...
 	usr, err := user.LookupGroup("gluster")
 	if err != nil {
-		return err
+		if _, ok := err.(user.UnknownGroupError); ok {
+			unknownGrp = true
+		} else {
+			return err
+		}
+	}
+	if err == nil {
```
Comment (rishubhjain) [inline glusterd2/gdctx/global.go]: @Madhu-1 This can be a simple else statement.

File: glusterd2/gdctx/global.go [hunk 2]
```diff
@@ -94,18 +94,35 @@ func GenerateLocalAuthToken() error {
...
 func protectAuthFile(authfile string) error {
+	var (
+		uGID       string
+		unknownGrp bool
+	)
 	usr, err := user.LookupGroup("gluster")
 	if err != nil {
-		return err
+		if _, ok := err.(user.UnknownGroupError); ok {
+			unknownGrp = true
```
Comment (aravindavk) [inline glusterd2/gdctx/global.go]: You can set `uGID = cuser.Gid` as default and avoid one extra variable(`unknownGrp`). Move `cuser, err := user.Current()` before `LookupGroup()`

Comment (Madhu-1) [reply glusterd2/gdctx/global.go]: done

**Comment by aravindavk** [comment]
> It doesn't appear that this change (and other changes -- but this on is convenient for me to comment on) has met the 24hr review period guidelines. This is a gentle reminder that we agreed to let changes remain open a while for wider community review and feedback.

24hr review period is only for non-trivial patches. Without this patch, glusterd2 will not start if "gluster" group doesn't exist before start(Regression from one of the patch). This is problem for other integrations who consume nightly builds.

**Comment by phlogistonjohn** [comment]
> 24hr review period is only for non-trivial patches. Without this patch, glusterd2 will not start if "gluster" group doesn't exist before start(Regression from one of the patch). This is problem for other integrations who consume nightly builds.

I thought that in the discussion we had discarded the trivial rule (it was too vague). On the other hand if this is need to prevent test runs from breaking I'm ok that it was "fast tracked". However, I would not have complained if that was noted in the PR description or a comment on the PR.

**Pull request merged**